pipelines/tasks/buildah-requirements.yaml

```
@@ -0,0 +1,54 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: buildah-requirements
+spec:
+  params:
+    - name: BASE_IMAGE
+      type: string
+    - description: Reference of the image buildah will produce.
+      name: IMAGE
+      type: string
+    - default: >-
+        registry.redhat.io/rhel8/buildah@sha256:6a68ece207bc5fd8db2dd5cc2d0b53136236fb5178eb5b71eebe5d07a3c33d13
+      description: The location of the buildah builder image.
+      name: BUILDER_IMAGE
+      type: string
+    - default: vfs
+      description: Set buildah storage driver
+      name: STORAGE_DRIVER
+      type: string
+    - default: "true"
+      description: >-
+        Verify the TLS on the registry endpoint (for push/pull to a non-TLS
+        registry)
+      name: TLSVERIFY
+      type: string
+    - default: oci
+      description: "The format of the built container, oci or docker"
+      name: FORMAT
+      type: string
+  workspaces:
+    - name: requirements
+      readonly: true
+      description: the list of pinned python package we install in the produced image
+  volumes:
+    - name: /var/lib/containers
+      emptyDir: {}
+  stepTemplate:
+    image: $(params.BUILDER_IMAGE)
+    env:
+      - name: STORAGE_DRIVER
+        value: $(params.STORAGE_DRIVER)
+      - name: BUILDAH_FORMAT
+        value: $(params.FORMAT)
+    volumeMounts:
+      - name: containers
+        mountPath: /var/lib/containers
+  steps:
+    - args: ["buildah", "from", "--name", "cnbi-image", "--tls-verify=$(params.TLSVERIFY)", "docker://$(params.BASE_IMAGE))"]
+    - args: ["buildah", "copy", "cnbi-image", "$(worspaces.requirements.path)/requirements-pinned.txt", "/tmp/"]
+    - args: ["buildah", "run", "cnbi-image", "--", "/usr/bin/pip", "install", "-r", "/tmp/requirements-pinned.txt"]
+    - args: ["buildah", "commit", "cnbi-image", "$(params.IMAGE)"]
+    - args: ["buildah", "push", "--tls-verify=$(params.TLSVERIFY)", "$(params.IMAGE)"]
```

I think the command should be separate from the arguments here (also suggesting to add names to the steps):

```suggestion    - args: ["from", "--name", "cnbi-image", "--tls-verify=$(params.TLSVERIFY)", "docker://$(params.BASE_IMAGE))"]
      command: [buildah]
      name: buildah-from
    - args: ["copy", "cnbi-image", "$(worspaces.requirements.path)/requirements-pinned.txt", "/tmp/"]
      command: [buildah]
      name: buildah-copy
    - args: ["run", "cnbi-image", "--", "/usr/bin/pip", "install", "-r", "/tmp/requirements-pinned.txt"]
      command: [buildah]
      name: buildah-run
    - args: ["commit", "cnbi-image", "$(params.IMAGE)"]
      command: [buildah]
      name: buildah-commit
    - args: ["push", "--tls-verify=$(params.TLSVERIFY)", "$(params.IMAGE)"]
      command: [buildah]
      name: buildah-push
```

---

> ```
> 2022/11/23 20:04:51 warning: unsuccessful cred copy: ".docker" from "/tekton/creds" to "/": unable to create destination directory: mkdir /.docker: permission denied
> time="2022-11-23T20:04:59Z" level=warning msg="Reading allowed ID mappings: reading subuid mappings for user \"1000710000\" and subgid mappings for group \"1000710000\": no subuid ranges found for user \"1000710000\" in /etc/subuid"
> time="2022-11-23T20:04:59Z" level=warning msg="Found no UID ranges set aside for user \"1000710000\" in /etc/subuid."
> time="2022-11-23T20:04:59Z" level=warning msg="Found no GID ranges set aside for user \"1000710000\" in /etc/subgid."
> Error during unshare(CLONE_NEWUSER): Function not implemented
> time="2022-11-23T20:04:59Z" level=error msg="parsing PID \"\": strconv.Atoi: parsing \"\": invalid syntax"
> time="2022-11-23T20:04:59Z" level=error msg="(Unable to determine exit status)"
> ```

That log worries me it looks like insufficient container perm. I think we can try to play with buildah isolation level but that need further investigation.